### ISSUE TITLE
Beeps are not sounded for a custom WinForms ProgressBar

### ISSUE THREAD

**Issue by NadezhdaDX** [nvaccess/nvda#15559]
### Steps to reproduce:

Hello, 
 
My name is Nadezhda. I'm a software engineer representing the WinForms team at DevExpress.  
We found that NVDA does not play a beep when our ProgressBarControl value is changed. The accessibility function for this control is designed based on the same technique as the standard ProgressBar. However, your tool plays a beep when the value is changed for only the standard control. Thus, we need a hint as to what exactly is required to implement this function.   
  
We attached a project ([WindowsFormsApp21.zip](https://github.com/nvaccess/nvda/files/12784431/WindowsFormsApp21.zip)) to demonstrate the differences in the behaviors. To reproduce it, you need to run this project and click on one of the controls. If you turn on NVDA at this moment and click the top control (a custom control), you do not get any sounds.  
In the bottom control (the standard control), you get a beep every time you click this control. 

### Actual behavior:

Beeps are sounded for only the bottom control.

### Expected behavior:

Beeps should be sounded for the top control as well.

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes

**Comment by seanbudd** [comment]
Welcome @NadezhdaDX ,

We are unable to process this issue as it stands.

Please fill out the issue template fully. What version of NVDA are you using to reproduce this issue? What version of Winforms?


Kind Regards,
NV Access Software Developers

**Comment by NadezhdaDX** [comment]
I use NVDA version 2023.2.
![nvda_S3Fw8UCv1O](https://github.com/nvaccess/nvda/assets/37904867/4f7d960a-2ca5-4918-a11d-96099d8b8391)

Regarding the WinForms project, it is reproduced in all the latest frameworks (.NET 4.8 and .NET 7).

**Comment by thgcode** [comment]
Hello!
I'm not a part of the NVDA or Winforms development teams but I think I found what might be the issue.
When the program updates the progress bar's value, it doesn't send any accessibility event to NVDA alerting that the progress bar's value was changed.
For NVDA to beep I think it listens to the valueChanged event on the program.
I changed the code on the onClick event to the following and it worked (tested both with Net Framework and Net 6):

```
        protected override void OnClick(EventArgs e) {
            base.OnClick(e);
            Value++;
            AccessibilityNotifyClients(AccessibleEvents.ValueChange, -1);

            Invalidate();
        }

```

Reference: OnClick handler for the button on winform's source code: https://github.com/dotnet/winforms/blob/951f1961898c1357b4484eb385378e7efd263f87/src/System.Windows.Forms/src/System/Windows/Forms/Button.cs#L238

I'm not sure if both the AccessibilityNotifyClients and the             AccessibilityObject.RaiseAutomationEvent calls are needed, I tested with only the first and it worked.
Hope this information can help you.
